### PR TITLE
 Externalising solinger for tcp options

### LINKED ISSUE #30005: Externalize soLinger for tcpOptions
The soLinger property  controls the trade-off between gracefully closing a connection and shutting it down. Since we are recommending setting the property to the customer we are documenting it.

### PR DESCRIPTION
**Pull request by ncpibm**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Fixes #30005

### CODE REVIEW

**Comment by ncpibm** [comment]
Fixes #30005

**Comment by ncpibm** [comment]
#libby #build

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=99580fb9-0fcd-403a-abc9-128732404f34

Target locations of links might be accessible only to IBM employees.

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=4f8d2594-07d6-4680-9484-9dbf82ea884f

Target locations of links might be accessible only to IBM employees.

**Comment by contbld** [comment]
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 2 product code files were changed.

**Comment by contbld** [comment]
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 2 product code files were changed.

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=99580fb9-0fcd-403a-abc9-128732404f34 completed successfully!

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=4f8d2594-07d6-4680-9484-9dbf82ea884f has errors/failures.

**Comment by ncpibm** [comment]
#libby #build

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=05889957-90eb-49e5-b6a5-97114804613e

Target locations of links might be accessible only to IBM employees.

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=dc0f813e-3fe3-4717-82a5-ebc589791f07

Target locations of links might be accessible only to IBM employees.

**Comment by contbld** [comment]
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 2 product code files were changed.

**Comment by contbld** [comment]
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 2 product code files were changed.

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=05889957-90eb-49e5-b6a5-97114804613e completed successfully!

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=dc0f813e-3fe3-4717-82a5-ebc589791f07 has errors/failures.

**Comment by ncpibm** [comment]
#build

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=e14c463a-8801-451b-9189-b89ee1209fb2

Target locations of links might be accessible only to IBM employees.

**Comment by contbld** [comment]
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 2 product code files were changed.

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=e14c463a-8801-451b-9189-b89ee1209fb2 has errors/failures.

**Comment by ncpibm** [comment]
#build

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=a868f0d6-1a93-4cf8-8a7f-1ed6d4b895fe

Target locations of links might be accessible only to IBM employees.

**Comment by contbld** [comment]
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 2 product code files were changed.

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=a868f0d6-1a93-4cf8-8a7f-1ed6d4b895fe has errors/failures.

**Comment by ncpibm** [comment]
#build

**Comment by ncpibm** [comment]
#build

**Comment by ncpibm** [comment]
#build

**Comment by ncpibm** [comment]
#build

**Comment by ncpibm** [comment]
#build

**Comment by ncpibm** [comment]
The RTC buiild ran successfully 
in the pipeline a test bucket is failing because of an existing defect unrelated to the change in this PR.

**Review by mrsaldana — APPROVED** [review]
LGTM

File: dev/com.ibm.ws.channelfw/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -66,3 +66,6 @@ tcp.maxOpenConnections.desc=Defines the maximum number of connections allowed to
 
 tcp.portOpenRetries=Port open retries
 tcp.portOpenRetries.desc=Number of retries to open a TCP/IP port during server startup.  There will be a one second delay between retries, until the opening is successful or the port open retry number is reached. 
+
+tcp.soLinger=SoLinger
+tcp.soLinger.desc=This parameter sets the delay for the connection while data is being transmitted before closing the socket, after a close request has been received. Set the value in the range of 0 to 65,536 to specify the delay period in seconds.Use -1 to disable this.
```
Comment (dmuelle) [inline dev/com.ibm.ws.channelfw/resources/OSGI-INF/l10n/metatype.properties:71]: Not sure why this was marked ID review not needed, but noticed on publishing the message that theres a typo and the message doesn't follow IBM Style guidance. If we can update the message, the following suggestion fixes the typo and adjusts for IBM Style

> This parameter sets the delay before closing the socket after a close request is received while data is transmitting. Set the value between 0 to 65,536 to specify the delay period in seconds. To disable this setting, set the value to -1 .

I can open a defect if needed- fyi @ramkumar-k-9286

Comment (ayoho) [reply dev/com.ibm.ws.channelfw/resources/OSGI-INF/l10n/metatype.properties:71]: @ncpibm @seshadhri-aswath Can you revisit @dmuelle's suggested change?

Comment (ncpibm) [reply dev/com.ibm.ws.channelfw/resources/OSGI-INF/l10n/metatype.properties:71]: HIi @ayoho  we have made the changes like @dmuelle suggested and merged it in this pr https://github.com/OpenLiberty/open-liberty/pull/30635

**Pull request merged**